provider/lxd/environ.go

```diff
@@ -292,3 +293,24 @@ func (env *environ) DeriveAvailabilityZones(
 	}
 	return []string{p.nodeName}, nil
 }
+
+// MaybeWriteLXDProfile implements environs.LXDProfiler.
+func (env *environ) MaybeWriteLXDProfile(pName string, put *charm.LXDProfile) error {
+	hasProfile, err := env.server.HasProfile(pName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if hasProfile {
+		logger.Debugf("lxd profile %q already exists, not written again", pName)
```

Comment:
This is great feedback for a user, so we can at least track this happened if debugging.